source/OutfitterPanel.cpp

```diff
@@ -106,6 +107,18 @@ void OutfitterPanel::Step()
 {
 	CheckRefill();
 	ShopPanel::Step();
+	if(GetUI()->IsTop(this))
+	{
+		Mission *mission = player.MissionToOffer(Mission::OUTFITTER);
+		// Special case: if the player somehow got to the outfitter before all
+		// landing missions were offered, they can still be offered here:
+		if(!mission)
+			mission = player.MissionToOffer(Mission::LANDING);
+		if(mission)
+			mission->Do(Mission::OFFER, player, GetUI());
+		else
+			player.HandleBlockedMissions(Mission::OUTFITTER, GetUI());
+	}
```

Comment:
It looks like you're adding basically the same code to `ShipyardPanel` and `OutfitterPanel`.
Have you considered putting this in the common superclass of those (`ShopPanel`, I think?) and having those call it in their `Step()` methods?

Reply:
I didn't look up the whole inheritance tree, but if they are the only two children can do that.

Reply:
Have this on local will push up in a second